store/postgres/src/store.rs

```diff
@@ -290,19 +290,27 @@ impl BasicStore for Store {
         }
 
         // Add order by filters to query
-        if let Some(order_attribute) = query.order_by {
+        if let Some((order_attribute, value_type)) = query.order_by {
             let direction = query
                 .order_direction
                 .map(|direction| match direction {
                     StoreOrder::Ascending => String::from("ASC"),
                     StoreOrder::Descending => String::from("DESC"),
                 }).unwrap_or(String::from("ASC"));
-
+            let cast_type = match value_type {
+                ValueType::BigInt => "::numeric".to_string(),
+                ValueType::Boolean => "::boolean".to_string(),
+                ValueType::Bytes => "".to_string(),
+                ValueType::Float => "::float".to_string(),
+                ValueType::ID => "".to_string(),
+                ValueType::Int => "::bigint".to_string(),
+                ValueType::String => "".to_string(),
+            };
```

Comment:
Here you can move the `to_string()` to after the match, so this line will be `}.to_string();`